domain/network/state/subnet.go

```diff
@@ -85,11 +85,23 @@ func (st *State) addSubnet(ctx context.Context, tx *sqlair.TX, subnetInfo networ
 	}
 	subnetUUID := subnetInfo.ID.String()
 
+	cidr, err := network.ParseCIDR(subnetInfo.CIDR)
```

Comment:
I don't agree with this. We shouldn't be parsing forms and validating here. This should have happened in the services layer. Ask for the form you need and get the heavy lifting up a layer.

Reply:
I've moved it up a layer.